packages/workflow/src/RoutingNode.ts

```diff
@@ -643,8 +643,15 @@ export class RoutingNode {
 		if (nodeProperties.routing) {
 			let parameterValue: string | undefined;
 			if (basePath + nodeProperties.name && 'type' in nodeProperties) {
+				// Extract value if it has extractValue defined or if it's a
+				// resourceLocator component. Resource locators are likely to have extractors
```

Comment:
I'm not familiar with the declarative design implementation. Is `RoutingNode.ts` only relevant for the declarative design? If so, for a node to use the record locator component, will it need to be refactored in the declarative way?

Reply:
Nope! `extractValue` can be passed into any call of `getNodeParameter`.